### PR TITLE
syncer: support start incremental task without configuring binlog-pos/gtid

### LINKED ISSUE #7393: support start incremental task without configuring binlog-pos/gtid
### Is your feature request related to a problem?

user don't have to start with exact binlog-pos/gtid

### Describe the feature you'd like

start a task without configuring binlog-pos/gtid，dm automatically starts from current pos/gtid

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #7393 

### What is changed and how it works?
- get newest binlog-pos/gtid from upstream

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
support start incremental task without configuring binlog-pos/gtid
```

### CODE REVIEW

File: dm/master/server_test.go [hunk 1]
```diff
@@ -2868,3 +2870,49 @@ func (t *testMaster) TestDashboardAddress(c *check.C) {
 	c.Assert(err, check.IsNil)
 	c.Assert(string(content), check.Matches, "[\\s\\S]*Web UI enabled[\\s\\S]*")
 }
+
+func (t *testMaster) TestGetNewestMeta(*check.C) {
```
Comment (lance6716) [inline dm/master/server_test.go:2874]:
```suggestion
func TestGetNewestMeta(t *testing.T) {
```

**Review by lance6716 — COMMENTED** [review]
and also test upstream has empty GTID set when GTID is enabled

File: dm/master/server_test.go (hunk 1 above)
Comment (GMHDBJD) [reply dm/master/server_test.go:2874]: use testSuite to avoid concurrency flaky test

**Comment by GMHDBJD** [comment]
> and also test upstream has empty GTID set when GTID is enabled

Does this meet the requirements? https://github.com/pingcap/tiflow/blob/d6dd20f0cc6378b602cda2c486a0396f54e3cd3f/dm/master/server_test.go#L2894
For empty GTID, I think that's same with all mode task like this issue https://github.com/pingcap/tiflow/issues/7037, cc @liumengya94

**Review by lance6716 — APPROVED** [review]

File: dm/master/server.go [hunk 2]
```diff
@@ -1384,6 +1384,31 @@ func parseSourceConfig(contents []string) ([]*config.SourceConfig, error) {
 	return cfgs, nil
 }
 
+// GetNewestMeta gets newest meta(binlog name, pos, gtid) from upstream.
+func GetNewestMeta(ctx context.Context, flavor string, dbConfig *config.DBConfig) (*config.Meta, error) {
```
Comment (buchuitoudegou) [inline dm/master/server.go:1388]: I would prefer `latest` to `newest`.

**Review by buchuitoudegou — APPROVED** [review]
lgtm
